lib/NetworkManager.js

```diff
@@ -327,6 +327,46 @@ class Request {
     });
   }
 
+  async mockResponse(response = {}) {
+    // DataURL's are not interceptable. In this case, do nothing.
+    if (this.url.startsWith('data:'))
```

Comment:
Should we throw here instead of failing silently?

Reply:
Good call.

Reply:
Left it as-is for the following code to work:

```js
page.on('request', request => request.respond({
  status: 404,
  body: Buffer.from('<div>Not Found!</div>')
});
```